api/src/main/java/bio/terra/tanagra/service/underlay/UnderlayConversion.java

```diff
@@ -406,16 +408,47 @@ public static Underlay convert(bio.terra.tanagra.proto.underlay.Underlay underla
             convert(hierarchyProto.getChildrenTable().getTableFilter(), columns));
       }
 
+      Column node =
+          retrieve(hierarchyProto.getPathsTable().getNode(), columns, "node", hierarchyProto);
+      Column path =
+          retrieve(hierarchyProto.getPathsTable().getPath(), columns, "path", hierarchyProto);
+      Hierarchy.PathsTable.Builder pathsTable =
+          Hierarchy.PathsTable.builder().node(node).path(path);
+
       Hierarchy hierarchy =
           Hierarchy.builder()
               .descendantsTable(descendantsTable.build())
               .childrenTable(childrenTable.build())
+              .pathsTable(pathsTable.build())
               .build();
+      // add a new attribute+mapping to the entity that contains the hierarchy path
+      Attribute hierarchyPathAttribute =
+          Attribute.builder()
+              .name("t_path_" + attribute.name())
```

Comment:
What does "t" in "t_path" stand for?

Reply:
Tanagra. I just wanted to namespace generated attributes and picked a prefix that's relatively short since there's a 31 character limit on the attribute name. I don't know of a hard reason for that limit, so we could also increase it and use a longer prefix like "tanagra_".